Sources/Tuvi.Core.DataStorage.Impl/DataStorage.cs

```diff
@@ -2199,5 +2214,31 @@ public Task DeleteAIAgentAsync(uint id, CancellationToken cancellationToken = de
                 connection.Delete<LocalAIAgent>(id);
             }, cancellationToken);
         }
+
+        public Task UpdateMessageProcessingResultAsync(Message message, string result, CancellationToken cancellationToken = default)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+
+            return WriteDatabaseAsync((db, ct) =>
+            {
+                var connection = db.Connection;
+                var item = connection.Find<Message>(message.Pk);
+                if (item == null)
```

Comment:
you can use `if (item is null)` or `var item = connection.Find<...>(...) ?? throw new DataBaseException(...)`